Sources/StreamChat/Utils/MulticastDelegate.swift

```diff
@@ -41,37 +41,37 @@ struct MulticastDelegate<T> {
     /// Sets the main delegate. If is nil, removes the main delegate.
     /// - Parameter mainDelegate: The main delegate.
     mutating func set(mainDelegate: T?) {
-        queue.async { [weak _mainDelegate] in
-            _mainDelegate?.removeAllObjects()
+        queue.sync {
```

Comment:
I'm curious if this will cause any performance issues 🤔 AFAIK this is called multiple times (More than it should)